core/src/main/scala/reftree/diagram/Animation.scala

```diff
@@ -83,6 +83,32 @@ object Animation {
   /** Create an animation builder from a sequence of starting values */
   def startWithSequence[A: ToRefTree](start: Seq[A]) = Builder(start.toVector)
 
+  private implicit class TakeUntilStream[T](s: Stream[T]) {
+    def takeWhileInclusive(p: T => Boolean): Stream[T] =
+      if (!s.isEmpty && p(s.head)) Stream.cons(s.head, s.tail takeWhileInclusive p)
+      else Stream.cons(s.tail.head, Stream.empty)
+  }
+
+  /** Create an animation builder from a seed until a fixpoint (f(x) == x) is reached */
+  def iterateUntilFixPoint[A: ToRefTree](seed: A)(iteration: (A => A))(
+    max: Int = 10) =
+    Builder(
+      Stream
+        .iterate(seed)(iteration)
+        .take(max)
+        .takeWhileInclusive(x => x != iteration(x))
+        .toVector)
+
+  /** Create an animation builder from a seed until a a given predicate yields true */
+  def iterateUntil[A: ToRefTree](seed: A)(iteration: (A => A))(
```

Comment:
I think an `iterateWhile` may be more useful, since one is free to choose between `.iterateWhile(F)(p).iterate(F)` or `.iterateWhile(F)(p).iterate(G)`.

Along that line of thinking, maybe the signature should be just `iterate(iteration: A ⇒ Option[A])`, which exhausts once the function returns `None`. What do you think? (Same comment about having separate limited and unlimited versions applies).